packages/sdks/output/qwik/src/blocks/columns/columns.tsx

```diff
@@ -0,0 +1,258 @@
+                },
+                key: {
+                  code: "index",
+                },
+              },
+              children: [
+                {
+                  "@type": "@builder.io/mitosis/node",
+                  name: "div",
+                  meta: {},
+                  scope: {},
+                  properties: {
+                    _text: "\n            ",
+                  },
+                  bindings: {},
+                  children: [],
+                },
+                {
+                  "@type": "@builder.io/mitosis/node",
+                  name: "RenderBlocks",
+                  meta: {},
+                  scope: {},
+                  properties: {},
+                  bindings: {
+                    blocks: {
+                      code: "column.blocks",
+                    },
+                  },
+                  children: [],
+                },
+                {
+                  "@type": "@builder.io/mitosis/node",
+                  name: "div",
+                  meta: {},
+                  scope: {},
+                  properties: {
+                    _text: "\n          ",
+                  },
+                  bindings: {},
+                  children: [],
+                },
+              ],
+            },
+          ],
+        },
+        {
+          "@type": "@builder.io/mitosis/node",
+          name: "div",
+          meta: {},
+          scope: {},
+          properties: {
+            _text: "\n    ",
+          },
+          bindings: {},
+          children: [],
+        },
+      ],
+    },
+  ],
+  hooks: {},
+  context: {
+    get: {},
+    set: {},
+  },
+  name: "Columns",
+  subComponents: [],
+  types: [
+    "type Column = {\n  blocks: any; // TODO: Implement this when support for dynamic CSS lands\n\n  width?: number;\n};",
+    "type StackColumnsAt = 'tablet' | 'mobile' | 'never';",
+  ],
+  interfaces: [
+    "export interface ColumnProps {\n  columns?: Column[]; // TODO: Implement this when support for dynamic CSS lands\n\n  space?: number; // TODO: Implement this when support for dynamic CSS lands\n\n  stackColumnsAt?: StackColumnsAt; // TODO: Implement this when support for dynamic CSS lands\n\n  reverseColumnsWhenStacked?: boolean;\n}",
+  ],
+  propsTypeRef: "ColumnProps",
+};
+function getGutterSize(props, state) {
+  return typeof props.space === "number" ? props.space || 0 : 20;
+}
+function getColumns(props, state) {
+  return props.columns || [];
+}
+function getWidth(props, state, index: number) {
+  const columns = getColumns(props, state);
+  return columns[index]?.width || 100 / columns.length;
+}
+function getColumnCssWidth(props, state, index: number) {
```

Comment:
Do we need to extract these functions to the top of the file instead of having them in the function body?

Reply:
Aka the component function body. Just bc mitosis tries to generate idiomatic code and this feels less idiomatic compared to normal react code that would just be in there and have normal function signatures (vs this idea of always having the props and state args)

this is obv a mitosis related comment but wanted to put in ckntext

Reply:
The issue here is that in all other frameworks there is a way to attach behavior (methods) to data. In Qwik functions can't be attached to data because they can't be serialized. This creates a problem, which I don't know how to solve. So for now I turn them into top lever functions. Because these functions capture props and state I need to pass these values directly.

@manucorporat do you have an idea on how to solve this?

Related issue: https://github.com/BuilderIO/qwik/issues/617#issuecomment-1157386259

Reply:
Now if Qwik can't take functions in the component body, then I think that is probably worth solving. But my understanding is the optimizer + useLexicalScope solves this

Reply:
also this is not well documented (my fault), but the idea of Mitosis outputs is they should be as idiomatic as possible (aka if someone "ejects" and decides to use the Qwik code directly, it should be as close to what people would have hand written as we can ideally get)

Reply:
Agree on "ejection"

So your suggestion will work for JSX because the function gets re-executed and hence the helper methods get re-created. But those methods will not be executable in say `useWatch$()` location or in `useMount$()` (Basically any place where they cross the `$` boundary. 

So from my point of view, we don't yet have an idiomatic way of doing that which can be used from all locations in Qwik, and that is what we need to solve. For now, it is the hack above, but I am not happy with the hack.

Reply:
Ah yes I see, functions in component body not currently supported, this works for me in the meantime 👍